### PR TITLE
Date Input / Time Picker

### PR DESCRIPTION
**Pull request by weili-govtech**
- misc refactor of date / time fields
- renamed date-helper as date-time-helper to better reflect its functionalities
  - updated `formatDateTime()` to accept format as string instead of DateTimeFormatter
  - updated to parse by date / time / datetime
- added default value story to time-picker
- updated packages
  - bumped design-system to alpha 10
  - resolve npm audit warnings with loader-utils
- updated storybook reference of design-system / font-awesome styles

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
 		"@js-joda/core": "^5.4.2",
 		"@js-joda/locale_en": "^4.8.6",
 		"@js-joda/timezone": "^2.15.0",
-		"@lifesg/react-design-system": "^1.0.0-alpha.7",
+		"@lifesg/react-design-system": "^1.0.0-alpha.10",
```
Comment (keithtxw) [inline package.json:30]: You might want to consider updating to the latest to test if it still works as expected

File: src/components/fields/date-input/date-input.tsx [hunk 2]
```diff
@@ -121,16 +112,27 @@ export const DateInput = (props: IGenericFieldProps<IDateInputSchema>) => {
 		} else {
```
Comment (keithtxw) [inline src/components/fields/date-input/date-input.tsx:112]: In line 108 
```
} else if (!day.length || !month.length || year.length !== 4) {
```
I am wondering if you should check if the day and month length <= 2 instead for more accuracy?

**Review by keithtxw — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (weili-govtech) [reply package.json]: ok I have updated it to alpha.14

File: src/components/fields/date-input/date-input.tsx [hunk 3]
```diff
@@ -114,23 +105,36 @@ export const DateInput = (props: IGenericFieldProps<IDateInputSchema>) => {
...
+				});
+			}
 		}
 	};
 
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const isValidDate = (value: string) => {
+		if (!value) return false;
```
Comment (keithtxw) [inline src/components/fields/date-input/date-input.tsx:131]: can we change this to `if (!value || value === "" || value === ERROR_MESSAGES.DATE.INVALID)`

**Review by keithtxw — CHANGES_REQUESTED** [review]

**Review by keithtxw — APPROVED** [review]

**Review by keithtxw — APPROVED** [review]

**Pull request merged**